File: VB/WindowsFormsApplication313/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo

Namespace WindowsFormsApplication313
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub
    End Class
    <ToolboxItem(True)> _
    Public Class CustomListBoxControl
        Inherits ListBoxControl

        Public Sub New()
            MyBase.New()

        End Sub
        Protected Overrides Function CreateViewInfo() As BaseStyleControlViewInfo
            Return New CustomListBoxViewInfo(Me)
        End Function
        Protected Overrides Function CreatePainter() As BaseControlPainter
            Return New CustomBaseListBoxPainter()
        End Function
    End Class

    Friend Class CustomListBoxViewInfo
        Inherits BaseListBoxViewInfo

        Public Sub New(ByVal owner As BaseListBoxControl)
            MyBase.New(owner)

        End Sub

        Protected Overrides Function CreateItemPainter() As BaseListBoxItemPainter
            Return New CustomBaseListBoxItemPainter()
        End Function
    End Class
    Public Class CustomBaseListBoxItemPainter
        Inherits ListBoxItemPainter

        Public Sub New()
            MyBase.New()

        End Sub
        Private indent As Integer = 20, customLeftIndent As Integer = 5
        Public Overrides Function GetHorzPadding(ByVal e As ListBoxItemObjectInfoArgs) As Integer
            'change width between items
            Dim result As Integer = MyBase.GetHorzPadding(e)

            Return result + indent
        End Function
        Public Overrides Sub DrawObject(ByVal e As DevExpress.Utils.Drawing.ObjectInfoArgs)
            MyBase.DrawObject(e)
        End Sub
        Protected Overrides Function CalcItemTextRectangle(ByVal e As ListBoxItemObjectInfoArgs) As Rectangle
            'change the left margin here

            Return Rectangle.Inflate(e.TextRect, -GetHorzTextIndent(e) - customLeftIndent, 0)
        End Function
        Public Overrides Function GetVertPadding(ByVal e As ListBoxItemObjectInfoArgs) As Integer
            Dim result As Integer = MyBase.GetVertPadding(e)
            Return result + indent
        End Function
    End Class
    Friend Class CustomBaseListBoxPainter
        Inherits BaseListBoxPainter

        Public Sub New()
            MyBase.New()

        End Sub
        'avoid painting the control's border
        Protected Overrides Sub DrawBorder(ByVal info As ControlGraphicsInfoArgs)
            ' base.DrawBorder(info);
        End Sub
    End Class
End Namespace

File: CS/WindowsFormsApplication313/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;

namespace WindowsFormsApplication313 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }
    }
    [ToolboxItem(true)]
    public class CustomListBoxControl : ListBoxControl {
        public CustomListBoxControl()
            : base() {

        }
        protected override BaseStyleControlViewInfo CreateViewInfo() {
            return new CustomListBoxViewInfo(this);
        }
        protected override BaseControlPainter CreatePainter() {
            return new CustomBaseListBoxPainter();
        }
    }

    class CustomListBoxViewInfo : BaseListBoxViewInfo {
        public CustomListBoxViewInfo(BaseListBoxControl owner)
            : base(owner) {

        }

        protected override BaseListBoxItemPainter CreateItemPainter() {
            return new CustomBaseListBoxItemPainter();
        }
    }
    public class CustomBaseListBoxItemPainter : ListBoxItemPainter {
        public CustomBaseListBoxItemPainter()
            : base() {

        }

        public override int GetHorzPadding(ListBoxItemObjectInfoArgs e) {
            //change width between items
            int result = base.GetHorzPadding(e);
            return result + 20;
        }
        public override void DrawObject(DevExpress.Utils.Drawing.ObjectInfoArgs e) {
            base.DrawObject(e);
        }
        protected override Rectangle CalcItemTextRectangle(ListBoxItemObjectInfoArgs e) {
            //change the left margin here
            return Rectangle.Inflate(e.TextRect, -fHorzTextIndent - 5, 0);
        }
        public override int GetVertPadding(ListBoxItemObjectInfoArgs e) {
            int result = base.GetVertPadding(e);
            return result + 20;
        }
    }
    class CustomBaseListBoxPainter : BaseListBoxPainter {
        public CustomBaseListBoxPainter()
            : base() {

        }
        //avoid painting the control's border
        protected override void DrawBorder(ControlGraphicsInfoArgs info) {
            // base.DrawBorder(info);
        }
    }
}
